### PR TITLE
Once again trying to fix the mingw build

### PR DESCRIPTION
**Pull request by askondro**
I'm not sure it there is seperate build for mingw and msvc. 
And If there is also a appveyor build mingw how can I integrate it into my fork?

### CODE REVIEW

**Comment by lynxlynxlynx** [comment]
There is, but we make a mingw build from linux:
https://github.com/ja2-stracciatella/ja2-stracciatella/actions/runs/11415454759/job/31765518302
Appveyor just uses windows.

What are you even trying to achieve?

**Comment by askondro** [comment]
I cannot build ja2-stracciatella on my MSYS2/UCRT64 console without linking issues with ntdll.

Only changing this line in CmakeList.txt fixes the linking of ja2.exe in MSYS2.

**Comment by askondro** [comment]
And what I'm trying to do is do the test you do on my own fork - before making a pull request

**Comment by lynxlynxlynx** [comment]
ok, the builds will run and I don't expect any problems, it's just another system library to link to.

**Comment by askondro** [comment]
Yeah, and it failed again. The question is why does it work at my machine but fails on the build machine.

**Comment by momoko-h** [comment]
@askondro , does the build succeed on your machine if you change line 279 in the root directory's CMakeList.txt from `LIST(APPEND JA2_LIBRARIES mingw_stdthreads)`to  `LIST(APPEND JA2_LIBRARIES mingw_stdthreads ntdll)`?

**Comment by lynxlynxlynx** [comment]
The dll seems to be missing from msys's linux bundle, that's how I read it.

**Comment by askondro** [comment]
It work like this way but if remove changes to the ja2-stracciatella\dependencies\lib-stracciatella\CMakeLists.txt

the following line 111:     set(ADDITIONAL_LIBS bcrypt ws2_32 shell32 advapi32 userenv gcc_eh pthread ntdll) -> set(ADDITIONAL_LIBS bcrypt ws2_32 shell32 advapi32 userenv gcc_eh pthread)

it will not link the ja2.exe

**Comment by momoko-h** [comment]
And try inserting a new line 17 "ntdll.dll" to install-dlls-mingw.cmake.in.

**Comment by askondro** [comment]
Like this?

`# install the dlls of the executables

set(CMAKE_CXX_COMPILER "@CMAKE_CXX_COMPILER@")
set(SDL2MAIN_LIBRARY "@SDL2MAIN_LIBRARY@")

# XXX add more core dlls when needed
# these dlls will never be installed
set(MINGW_CORE_DLLS
    "advapi32.dll"
    "bcrypt.dll"
    "comctl32.dll"
    "comdlg32.dll"
    "gdi32.dll"
    "imm32.dll"
    "kernel32.dll"
    "msvcrt.dll"
    "ole32.dll"
    "oleaut32.dll"
    "setupapi.dll"
    "shell32.dll"
    "user32.dll"
    "userenv.dll"
    "version.dll"
    "winmm.dll"
    "ws2_32.dll"
    "ntdll.dll"
)`

**Comment by momoko-h** [comment]
Yes, but that list is ordered alphabetically, please keep it that way.

**Comment by askondro** [comment]
I will until CI for mingw64 finishes. Hopefully ...

**Comment by askondro** [comment]
Passes al builds now :)

**Comment by askondro** [comment]
This is related to issue #1917

File: cmake/install-dlls-mingw.cmake.in [hunk 1]
```diff
@@ -23,6 +24,7 @@ set(MINGW_CORE_DLLS
     "version.dll"
     "winmm.dll"
     "ws2_32.dll"
+
```
Comment (lynxlynxlynx) [inline cmake/install-dlls-mingw.cmake.in:27]: Can you fix these two? Misalignment and a needless empty line.

Comment (askondro) [reply cmake/install-dlls-mingw.cmake.in]: Yeah somehow it really depends on the editor. Notepad++ did not show me any difference.

Comment (lynxlynxlynx) [reply cmake/install-dlls-mingw.cmake.in]: It depends on the tab width you have set.

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**